### PR TITLE
Apple2: rewrite readdir + closedir + rewinddir to assembly

### PR DESCRIPTION
**Pull request by colinleroy**
177 bytes gained for readdir+closedir
130 bytes for rewinddir

### CODE REVIEW

File: libsrc/apple2/opendir.s [hunk 1]
```diff
@@ -154,3 +115,46 @@
...
+; Read a directory for the DIR* pointer in ptr1
+; Return with carry clear on success
+read_dir_block_ptr1:
+        ; Push ptr1, read will destroy it
+        jsr       pushptr1
+
+        ldy       #DIR::FD
+        lda       (ptr1),y
+
+        jsr       pusha0      ; push fd for read
```
Comment (oliverschmidt) [inline libsrc/apple2/opendir.s:128]: Start comment in uppercase.

**Comment by oliverschmidt** [comment]
I believe this PR obsoletes https://github.com/cc65/cc65/blob/master/libsrc/apple2/dir.h.

**Comment by oliverschmidt** [comment]
@colinleroy Please delete `libsrc/apple2/dir.h`.

**Comment by colinleroy** [comment]
> @colinleroy Please delete `libsrc/apple2/dir.h`.

Hi! Can't do that yet, there is still rewinddir.c todo. I was planning on another PR for it if that's OK with you?

**Comment by colinleroy** [comment]
(By the way there's a bug in rewinddir.c, I've pushed a very little commit addressing it before rewriting it)

**Comment by colinleroy** [comment]
Actually, here you are, rewinddir included :)

**Comment by oliverschmidt** [comment]
Great to see the Apple II dir stuff being consistently asm now 😆.

I'm personally fine with merging the PR. However, GH tells me
{6F74780D-F885-4684-86DB-221B67125741}
so I ask someone who knows what that means to take it form here.

**Comment by colinleroy** [comment]
Ah. I'll do it tomorrow :)

**Comment by colinleroy** [comment]
Voilà, I remade the branch with two clean commits. Should be mergeable now :)

**Pull request merged**